### PR TITLE
Multiple secret sources docs

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

Documentation changes to update guides related to storing secrets in Vault and/or Kubernetes.

This includes updating the Cert-Manager guide with info on how to use Vault PKI as a CA for C-M with all other secrets in Vault

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/main/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/4863

### CODE REVIEW

File: docs/content/installation/advanced_configuration/vault_secrets.md [hunk 1]
```diff
@@ -76,20 +77,21 @@ An example using AWS IAM auth might look like the following:
        xdsBindAddr: 0.0.0.0:9977
      kubernetesArtifactSource: {}
      kubernetesConfigSource: {}
-     # Delete or comment out the existing
-     # kubernetesSecretSource or directorySecretSource field
+     # Delete or comment out the existing \*SecretSource field
```
Comment (jbohanon) [inline docs/content/installation/advanced_configuration/vault_secrets.md:80]: this didn't render correctly. going to check locally before committing these changes

```suggestion
     # Delete or comment out the existing *SecretSource field
```

File: docs/content/installation/advanced_configuration/vault_secrets.md [hunk 2]
```diff
@@ -43,24 +43,25 @@ Edit the `default` settings resource so Gloo Edge reads and writes secrets using
...
```
Comment (jbohanon) [inline docs/content/installation/advanced_configuration/vault_secrets.md:46]: this didn't render correctly. going to check locally before committing these changes

```suggestion
     # Delete or comment out the existing *SecretSource field
```

File: docs/content/guides/integrations/cert_manager/_index.md [hunk 3]
```diff
@@ -63,437 +42,446 @@ export RS='{ "Changes": [{"Action": "UPSERT", "ResourceRecordSet":{"ResourceReco
...
+     discoveryNamespace: gloo-system
+     gateway:
+       validation:
+         alwaysAccept: true
+         proxyValidationServerAddr: gloo:9988
+     gloo:
+       xdsBindAddr: 0.0.0.0:9977
+     kubernetesArtifactSource: {}
+     kubernetesConfigSource: {}
+     # Delete or comment out the existing \*SecretSource field
```
Comment (jbohanon) [inline docs/content/guides/integrations/cert_manager/_index.md:471]: this didn't render correctly. going to check locally before committing these changes

```suggestion
     # Delete or comment out the existing *SecretSource field
```

File: docs/content/guides/integrations/cert_manager/_index.md [hunk 4]
```diff
@@ -63,437 +42,446 @@ export RS='{ "Changes": [{"Action": "UPSERT", "ResourceRecordSet":{"ResourceReco
...
+
+You just confirmed that the service is accessible over the HTTPS port and that Envoy gets the certificate from Let's Encrypt!
+
+## Use HashiCorp Vault as a Certificate Authority {#vault-ca}
+
+Cert-manager supports using HashiCorp Vault as a CA. For Gloo Edge to use the certificates from Vault, you must set up a Vault instance as the CA. Then, set up cert-manager to use that Vault instance as an `Issuer`.
+
+1. [Set up Vault as a CA by using the PKI secrets engine to generate certificates](https://developer.hashicorp.com/vault/docs/secrets/pki).
+2. [Create a cert-manager `Issuer` for the Vault CA](https://cert-manager.io/docs/configuration/vault/).
+3. If you use Vault to store other, non-TLS secrets along with the TLS certificates, then configure you default Gloo Edge Settings.
```
Comment (jbohanon) [inline docs/content/guides/integrations/cert_manager/_index.md:447]: The TLS secrets are stored in Kubernetes for this configuration
```suggestion
3. If you use Vault to store other, non-TLS secrets, then configure your default Gloo Edge Settings.
```

**Review by artberger — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**